### PR TITLE
Featured voting

### PR DESCRIPTION
**Pull request by jkbktl**
- [x] load initial SNT amount from proposal vote (contract?)
- [x] load cooldown period for communities which were featured and cannot be voted for again (contract?)
- [x] verification
- [x] testing

### CODE REVIEW

File: packages/DApp/src/components/card/VoteConfirmModal.tsx [hunk 1]
```diff
@@ -27,7 +30,12 @@ export function VoteConfirmModal({ community, selectedVote, setShowModal }: Vote
       <EtherscanLink href="#" target="_blank">
         View on Etherscan
       </EtherscanLink>
-      <ConfirmBtn onClick={() => setShowModal(false)}>
+      <ConfirmBtn
+        onClick={() => {
+          setShowModal(false)
+          history.go(0)
```
Comment (jkbktl) [inline packages/DApp/src/components/card/VoteConfirmModal.tsx:36]: Added reload of the page, so user can see newly proposed votings.

File: packages/DApp/src/pages/Featured.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
```
Comment (jkbktl) [inline packages/DApp/src/pages/Featured.tsx:1]: Newly created page, so we can show featured communities.

File: packages/DApp/src/components/NotificationsList.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ export function NotificationsList() {
       {notifications.map((notification) => {
         if ('receipt' in notification) {
           return notification.receipt.logs.map((log) => {
+            // this needs to be updated so it takes into account also interface of featuredVotingContract
             const parsedLog = votingContract.interface.parseLog(log)
```
Comment (jkbktl) [inline packages/DApp/src/components/NotificationsList.tsx:18]: Here is the issue with NotificationList, it accept list of notifications which can be emmited by both contracts (voting and featuredVoting), but there is currently no way how to differentiate notification item and therefore parse it by parser with correct interface. If notification item is parsed by wrong parser it throws an exception and crashes whole page.

**Comment by jkbktl** [comment]
Guys, it's ready for review, please check it out when you have a bit of time. cc @osmaczko @prichodko

**Comment by osmaczko** [comment]
> Guys, it's ready for review, please check it out when you have a bit of time. I cannot add you as reviewers for some reason. cc @osmaczko @prichodko

Cool! I am on it. Could you please check tests as they are failing on CI?

**Review by osmaczko — COMMENTED** [review]
Overall looks good, thanks for implementation. However, I did spot a few minor issues:

1. Once the directory vote is finalized the view is not updated

https://github.com/status-im/community-dapp/assets/33099791/c57b4d32-ec6d-494a-8a4d-1120def969c8

2. It should not be possible to vote more than once for featured community from the same account

https://github.com/status-im/community-dapp/assets/33099791/7aa45a88-5b05-492c-aa5b-2b66e8f2233a

**Review by osmaczko — CHANGES_REQUESTED** [review]
Thanks. It mostly works as expected, I identified few issues:

1) (blocker) Peers should be able to vote for more than one community

https://github.com/status-im/community-dapp/assets/33099791/0153d10f-b883-450d-b126-1271f15e2088

2) (blocker) "Verify Weekly featured" doesn't work for consecutive votings (works only for first voting)

https://github.com/status-im/community-dapp/assets/33099791/30bfd3bb-976d-449b-8888-fdf69747c805

3) (minor - will create a separate issue for that) directory page refreshes itself after each vote

**Comment by jkbktl** [comment]
Peers can vote for more than one community, but they can't vote multiple times for one community?

**Comment by osmaczko** [comment]
> Peers can vote for more than one community, but they can't vote multiple times for one community?

Yes, exactly.

**Review by osmaczko — APPROVED** [review]
Tested, core functionality works fine :muscle: 

I'll create separate issues for minor stuff as followup.

**Comment by osmaczko** [comment]
I gave it a second try and identified one critical issue, although I don't want to block this PR any longer. We should address it in followup.

https://github.com/status-im/community-dapp/assets/33099791/649cc729-644a-4010-8e10-83e1ac69fe3e

1) The featured communities list is wrong, `From contract for portal` should not make it into the list, as it has the lowest amount of SNT. What I think happened is either `castVotes` didn't propagate all the votes to the contract or the topN calculation is done wrongly by the contract itself. I think it is rather the former because I just double checked topN calculation in contract, also it is covered with tests.

2) The button says "Finalize Weekly featured" instead of "Verify Weekly featured", this seems to be state update issue.

File: packages/contracts/package.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
     "chai": "^4.2.0",
     "eslint": "^8.36.0",
     "ethers": "5.4.1",
-    "hardhat": "^2.13.0",
+    "hardhat": "^2.16.1",
```
Comment (felicio) [inline packages/contracts/package.json:37]: see https://github.com/NomicFoundation/hardhat/issues/3877#issuecomment-1610103284

**Comment by jkbktl** [comment]
@osmaczko let's merge it then (I don't have permissions)

**Pull request merged**